zmon_aws_agent/postgresql.py

```diff
@@ -8,27 +8,41 @@
 from zmon_aws_agent.aws import entity_id
 from zmon_aws_agent.common import call_and_retry
 
+from opentracing_utils import trace, extract_span_from_kwargs
+from opentracing.ext import tags as ot_tags
+
 
 logger = logging.getLogger(__name__)
 
 POSTGRESQL_DEFAULT_PORT = 5432
 
 
+@trace(pass_span=True, tags={'aws': 'postgres'})
 def list_postgres_databases(*args, **kwargs):
     try:
-        conn = psycopg2.connect(*args, **kwargs)
-        cur = conn.cursor()
-        cur.execute("""
+        query = """
             SELECT datname
               FROM pg_database
              WHERE datname NOT IN('postgres', 'template0', 'template1')
-        """)
+        """
+        current_span = extract_span_from_kwargs(**kwargs)
+        current_span.set_tag(ot_tags.PEER_ADDRESS,
+                             'psql://{}:{}'.format(kwargs.get('host'), kwargs.get('port')))
+        current_span.set_tag(ot_tags.DATABASE_INSTANCE, kwargs.get('dbname'))
+        current_span.set_tag(ot_tags.DATABASE_STATEMENT, query)
+
+        conn = psycopg2.connect(*args, **kwargs)
+        cur = conn.cursor()
+        cur.execute()
         return [row[0] for row in cur.fetchall()]
-    except Exception:
+    except Exception as e:
+        current_span.set_tag('error', True)
+        current_span.log_kv({'exception': str(e)})
```

Comment:
May be this is a good candidate for ``traceback.format_exc()`` instead of ``str(e)``